lib/mysql_framework/sql_condition.rb

```diff
@@ -3,18 +3,31 @@
 module MysqlFramework
 # This class is used to represent a Sql Condition for a column.
   class SqlCondition
+    NIL_COMPARISONS = ['IS NULL', 'IS NOT NULL'].freeze
+
     # This method is called to get the value of this condition for prepared statements.
     attr_reader :value
 
-    def initialize(column:, comparison:, value:)
+    def initialize(column:, comparison:, value: nil)
       @column = column
       @comparison = comparison
+
+      raise ArgumentError, "Cannot set value when comparison is #{comparison}" if nil_comparison? && value != nil
```

Comment:
does this not also imply that you shouldn't use `nil` when the operator is not a `NIL_COMPARISONS`?
i.e.
```ruby
  raise ArgumentError, 'value is nil but operator is not' if ` if nil_comparison? == false && value.nil?
```

Reply:
I think @rwadstein makes a good point. When you use `column: 'column_name', comparison: '=', value: nil`, the following SQL would be generated `... WHERE column_name = nil`

Reply:
Addressed in 82d9d81